docs/source/API/core/printf.rst

```diff
@@ -0,0 +1,17 @@
+``Kokkos::printf``
+=================
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+Defined in header ``<Kokkos_Core.hpp>``
+
+.. code-block:: cpp
+
+    template <typename... Args>
+    KOKKOS_FUNCTION int printf(const char* format, Args... args);
+
+For the Serial backend, ``Kokkos::printf`` is implemented in terms of ``std::printf`` printing the data specified in ``format`` and ``args...`` to ``stdout``.
+Returns the number of characters written if successful or a negative value if an error encountered.
+
+The implementation for other backends emulates this behavior.
```

Comment:
By the "emulates this behavior," do you mean in all of the non-serial cases across all of the backends?

Final ask:  would it be possible for you to give a examples of serial and non-serial cases to see the difference in what you see in `stdout`?

Reply:
I don't have an example. `printf` behaves the same for all implementations right now to the bets of my knowledge but we don't necessarily want to promise that there might not be any deviations for device implementations.

Reply:
Group comment:

- Revisit once code PR is merged